core/src/main/java/org/infinispan/interceptors/impl/CacheLoaderInterceptor.java

```diff
@@ -498,76 +584,120 @@ long calculateTimeoutSeconds() {
          return Math.round(Math.log1p(size) * 10);
       }
 
+      private Map.Entry<K, V> toEntry(Object obj) {
+         if (obj instanceof Map.Entry) {
+            return (Map.Entry) obj;
+         } else {
+            return null;
```

Comment:
The cast would work with`null` as well, what kind of objects are we filtering out here?

Reply:
👍

Reply:
Actually it has to be like this. The problem is the user can pass anything since method declaration is `Object`